### PR TITLE
MAP-1475

### PR DESCRIPTION
**Pull request by michalnawrockidj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/LocationApiClient.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+@Component
+class LocationApiClient(@Qualifier("locationApiWebClient") private val webClient: WebClient) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun getNonResidentialLocationForPrison(prisonId: String, usageType: String): List<LocationDetails> {
+    return webClient.get()
+      .uri("/locations/prison/$prisonId/non-residential-usage-type/$usageType")
```
Comment (GurnankCheema) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/LocationApiClient.kt:19]: nice

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/LocationService.kt [hunk 2]
```diff
@@ -14,6 +16,11 @@ class LocationService(
       .filter { it.locationType == "VIDE" }
       .map { LocationIdAndDescription(it.locationId, it.userDescription ?: it.description) }
 
+  fun getVideoLinkRoomsForPrisonFromLocation(agencyId: String): List<LocationInsidePrisonIdAndDescription> =
```
Comment (GurnankCheema) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/LocationService.kt:19]: maybe add 'Api' to end of func name?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/integration/VideoLinkBookingRoomsIntegrationTest.kt [hunk 3]
```diff
@@ -22,6 +22,22 @@ class VideoLinkBookingRoomsIntegrationTest : IntegrationTest() {
...
+    locationApiMockServer.stubGetAllLocationForPrison("WSI")
+
+    webTestClient.get()
+      .uri("/location/video-link-rooms/WSI")
+      .headers(setHeaders())
+      .exchange()
+      .expectStatus().isOk
+      .expectBody()
+      .consumeWith(System.out::println)
+      .jsonPath("[0].locationId").isEqualTo("5e449d34-446d-47de-9e93-98190a8fbd80")
```
Comment (GurnankCheema) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/integration/VideoLinkBookingRoomsIntegrationTest.kt:37]: not sure about .consumeWith(System.out::println) ?

**Review by GurnankCheema — APPROVED** [review]
LGTM but with minor comments

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/VideoLinkBookingRoomsController.kt [hunk 4]
```diff
@@ -33,4 +34,19 @@ class VideoLinkBookingRoomsController(
     @PathVariable("agencyId")
     agencyId: String,
   ): List<LocationIdAndDescription> = locationService.getVideoLinkRoomsForPrison(agencyId)
+
+  @GetMapping(
+    path = ["/location/video-link-rooms/{agencyId}"],
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/VideoLinkBookingRoomsController.kt:39]: can you change this to prisonId

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/VideoLinkBookingRoomsController.kt [hunk 5]
```diff
@@ -33,4 +34,19 @@ class VideoLinkBookingRoomsController(
...
+    produces = [MediaType.APPLICATION_JSON_VALUE],
+  )
+  @ResponseStatus(HttpStatus.OK)
+  @Operation(
+    description = "List of all the Video Link Booking rooms in the prison.",
+    summary = "getVideoLinkBookingRooms",
+  )
+  fun getVideoLinkBookingRoomsFromLocationApi(
+    @Parameter(description = "The prison", required = true)
+    @PathVariable("agencyId")
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/VideoLinkBookingRoomsController.kt:49]: ditto

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/LocationService.kt [hunk 6]
```diff
@@ -14,6 +16,11 @@ class LocationService(
       .filter { it.locationType == "VIDE" }
       .map { LocationIdAndDescription(it.locationId, it.userDescription ?: it.description) }
 
+  fun getVideoLinkRoomsForPrisonFromLocationApi(agencyId: String): List<LocationInsidePrisonIdAndDescription> =
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/LocationService.kt:19]: Change to prisonId

**Review by Mjwillis — APPROVED** [review]
Just change from agencyId to prisonId

**Review by Mjwillis — COMMENTED** [review]
agencyId=prisonId

**Review by GurnankCheema — APPROVED** [review]
LGTM

**Pull request merged**